operators/constellation-node-operator/internal/cloud/azure/client/nodeimage.go

```diff
@@ -27,12 +28,34 @@ func (c *Client) GetNodeImage(ctx context.Context, providerID string) (string, e
 	if resp.Properties == nil ||
 		resp.Properties.StorageProfile == nil ||
 		resp.Properties.StorageProfile.ImageReference == nil ||
-		resp.Properties.StorageProfile.ImageReference.ID == nil && resp.Properties.StorageProfile.ImageReference.CommunityGalleryImageID == nil {
+		resp.Properties.StorageProfile.ImageReference.ID == nil &&
+			resp.Properties.StorageProfile.ImageReference.CommunityGalleryImageID == nil &&
+			resp.Properties.StorageProfile.ImageReference.Publisher == nil &&
+			resp.Properties.StorageProfile.ImageReference.Offer == nil &&
+			resp.Properties.StorageProfile.ImageReference.SKU == nil &&
+			resp.Properties.StorageProfile.ImageReference.Version == nil {
 		return "", fmt.Errorf("node %q does not have valid image reference", providerID)
 	}
+
+	// Marketplace Image is used, format it to an URI and return it.
+	if resp.Properties.StorageProfile.ImageReference.Publisher != nil &&
+		resp.Properties.StorageProfile.ImageReference.Offer != nil &&
+		resp.Properties.StorageProfile.ImageReference.SKU != nil &&
+		resp.Properties.StorageProfile.ImageReference.Version != nil {
+		return mpimage.AzureMarketplaceImage{
```

Comment:
nit: You could skip having to do all these checks if you run the marketplace image check last, i.e. after checking if Image ID or Community Gallery ID are set.
Because in that case the marketplace values have to be set.